test/spec/SearchParameters/removeXFacetRefinement.js

```diff
@@ -0,0 +1,304 @@
+    var state = new SearchParameters({
+      facets: ['attribute'],
+      facetsExcludes: {
+        attribute: ['value']
+      }
+    });
+
+    expect(state.removeExcludeRefinement('attribute')).toEqual(
+      new SearchParameters({
+        facets: ['attribute'],
+        facetsExcludes: {
+          attribute: []
+        }
+      })
+    );
+  });
+
+  test('removeExcludeRefinement(attribute, value)', function() {
+    var state = new SearchParameters({
+      facets: ['attribute'],
+      facetsExcludes: {
+        attribute: ['value', 'value2']
+      }
+    });
+
+    expect(state.removeExcludeRefinement('attribute', 'value')).toEqual(
+      new SearchParameters({
+        facets: ['attribute'],
+        facetsExcludes: {
+          attribute: ['value2']
+        }
+      })
+    );
+  });
+
+  test('removeExcludeRefinement(attribute, lastValue)', function() {
+    var state = new SearchParameters({
+      facets: ['attribute'],
+      facetsExcludes: {
+        attribute: ['value']
+      }
+    });
+
+    expect(state.removeExcludeRefinement('attribute', 'value')).toEqual(
+      new SearchParameters({
+        facets: ['attribute'],
+        facetsExcludes: {
+          attribute: []
+        }
+      })
+    );
+  });
+});
+
+describe('removeTagRefinement', function() {
+  test('removeTagRefinement(tag)', function() {
+    var state = new SearchParameters({
+      tagRefinements: ['tag', 'tag2']
+    });
+
+    expect(state.removeTagRefinement('tag')).toEqual(
+      new SearchParameters({
+        tagRefinements: ['tag2']
+      })
+    );
+  });
+
+  test('removeTagRefinement(lastTag)', function() {
+    var state = new SearchParameters({
+      tagRefinements: ['lastTag']
+    });
+
+    expect(state.removeTagRefinement('lastTag')).toEqual(
+      new SearchParameters({
+        tagRefinements: []
+      })
+    );
+  });
+});
+
+describe('removeHierarchicalFacetRefinement', function() {
+  // NOTE: removeHierarchicalFacetRefinement only allows to remove a whole attribute
+  test('removeHierarchicalFacetRefinement(attribute)', function() {
+    var state = new SearchParameters({
+      hierarchicalFacets: [{name: 'attribute'}],
+      hierarchicalFacetsRefinements: {
+        attribute: ['value', 'value2']
+      }
+    });
+
+    expect(state.removeHierarchicalFacetRefinement('attribute')).toEqual(
+      new SearchParameters({
+        hierarchicalFacets: [{name: 'attribute'}],
+        hierarchicalFacetsRefinements: {
+          attribute: []
+        }
+      })
+    );
+  });
+});
+
+describe('removeNumericRefinement', function() {
+  test('removeNumericRefinement(attribute)', function() {
+    var state = new SearchParameters({
+      numericRefinements: {
+        attribute: {
+          '>=': [100]
+        }
+      }
+    });
+
+    expect(state.removeNumericRefinement('attribute')).toEqual(
+      new SearchParameters({
+        numericRefinements: {
+          attribute: {
+            '>=': []
+          }
```

Comment:
This might be unexpected since there’s no “removeNumericFacet” to remove it fully

Reply:
This works as expected, since we exclude empty operators in InstantSearch in the getRefinements reimplementation there

Reply:
We don't have a function `removeNumericFacet` but it's our responsibility to remove omit it inside `dispose`.